src/settings/storeconf.cpp

```diff
@@ -161,6 +191,10 @@ void StoreConf::writeJson(sessionMode mode)
                 root_obj.insert(it.key(), s.value(it.key()).toString());
             break;
         }
+        default:
```

Comment:
I didn't see it in the previous pull request. 
In the fucntion writeJson() (119+ line) you do the same operation 4 times!
``` 
        if (s.contains("cppExtentionsList"))
        {
            QString strExtentions;
            QStringList extentionList = s.value("cppExtentionsList").toStringList();
            for (int i = 0; i < extentionList.size(); ++i)
            {
                if(i != 0)
                    strExtentions += ";";
                strExtentions += extentionList.at(i);
            }
            s.setValue("cppExtentions", strExtentions);
        }
        if (s.contains("cppCompilersList"))
        {
            QString strCompilers;
            QStringList compilersList = s.value("cppCompilersList").toStringList();
            for (int i = 0; i < compilersList.size(); ++i)
            {
                if(i != 0)
                    strCompilers += ";";
                strCompilers += compilersList.at(i);
            }
            s.setValue("cppCompilers", strCompilers);
        }
        if (s.contains("cppLibraryHeadersList"))
        {
            QString strHeaders;
            QStringList headersList = s.value("cppLibraryHeadersList").toStringList();
            for (int i = 0; i < headersList.size(); ++i)
            {
                if(i != 0)
                    strHeaders += ";";
                strHeaders += headersList.at(i);
            }
            s.setValue("cppLibraryHeaders", strHeaders);
        }
        if (s.contains("cppLibraryBinarysList"))
        {
            QString strBinarys;
            QStringList binarysList = s.value("cppLibraryBinarysList").toStringList();
            for (int i = 0; i < binarysList.size(); ++i)
            {
                if(i != 0)
                    strBinarys += ";";
                strBinarys += binarysList.at(i);
            }
            s.setValue("cppLibraryBinarys", strBinarys);
        }
    }```

use only one fucntion!

`void func(const QString &nameList)
        {
            QString strName;
            QStringList nameList = s.value(nameList).toStringList();
            for (int i = 0; i < nameList.size(); ++i)
            {
                if(i != 0)
                    strName += ";";
                strName += nameList.at(i);
            }
            s.setValue(name, strName);
        }`